File: frontend/src/types/question.ts
/**
 * 質問データの型定義
 * Requirements: 9.1 - 質問データ管理
 */

/** 選択肢（single_choice型） */
export interface Choice {
  id: string;
  label: string;
}

/** チェックボックス選択肢（multi_select型） */
export interface MultiSelectOption {
  id: string;
  label: string;
  tags: string[];
}

/** 質問タイプ */
export type QuestionType = 'single_choice' | 'multi_select';

/** 質問 */
export interface Question {
  id: string;
  text: string;
  category_id: string;
  question_type: QuestionType;
  source_reference: string;
  /** single_choice型の選択肢 */
  choices: Choice[];
  /** multi_select型の選択肢 */
  options: MultiSelectOption[];
  /** multi_select: 最低選択数 (0=制限なし) */
  min_select: number;
  /** multi_select: 最大選択数 (0=制限なし) */
  max_select: number;
}

/** カテゴリ */
export interface Category {
  id: string;
  name: string;
  order: number;
  questions: Question[];
}

File: backend/app/models/question.py
"""質問データモデル: 選択肢・質問・カテゴリ"""

from typing import Literal

from pydantic import BaseModel, Field


class Choice(BaseModel):
  """質問の選択肢（4択のうちの1つ）"""

  id: str
  label: str = Field(..., max_length=100)


class MultiSelectOption(BaseModel):
  """チェックボックス型質問の選択肢"""

  id: str
  label: str = Field(..., max_length=100)
  tags: list[str] = Field(default_factory=list)
  """選択時にlexical_tagsに追加されるタグ群"""


class Question(BaseModel):
  """個別質問定義

  question_type:
    - "single_choice": 従来の4択+Other（スコアリング対象）
    - "multi_select": チェックボックス複数選択（タグ収集用、スコアリング非対象）

  single_choice 型: choices フィールドを使用
  multi_select 型: options フィールドを使用
  """

  id: str
  text: str = Field(..., max_length=200)
  category_id: str
  question_type: Literal["single_choice", "multi_select"] = "single_choice"
  choices: list[Choice] = Field(default_factory=list)
  """single_choice型の選択肢（4つ固定）"""
  options: list[MultiSelectOption] = Field(default_factory=list)
  """multi_select型の選択肢（5〜15個、複数選択可）"""
  min_select: int = 0
  """multi_select型: 最低選択数（0=制限なし）"""
  max_select: int = 0
  """multi_select型: 最大選択数（0=制限なし）"""
  source_reference: str = ""


class Category(BaseModel):
  """質問カテゴリ

  Business OS → Communication → Lifestyle/Hobbies → Interests & Preferences の
  固定順序で質問をグルーピングする。
  """

  id: str
  name: str
  order: int
  questions: list[Question] = Field(default_factory=list)

